### ISSUE TITLE
Allow multi-valued tags

### ISSUE THREAD

**Issue by DeltaWhy** [inspec/inspec#1578]
We'd like to be able to add tags with multiple values, e.g.:

```ruby
control 'V-72841' do 
...
  tag cci: 'CCI-000382'
  tag nist: 'CM-7 b'
  tag nist_rev: '4'

  tag cci: 'CCI-001762'
  tag nist: 'CM-7 (1) (b)'
  tag nist_rev: '4'
...
end
```

This should result in a tags structure that looks like:
```ruby
{cci: ['CCI-000382', 'CCI-001762'], nist: ['CM-7 b', 'CM-7 (1) (b)'], nist_rev: ['4', '4']}
```

We could do this now by passing an array to the 'tag' command but this seems like a more readable way to accomplish it. Thoughts?

CC @aaronlippold

**Comment by adamleff** [comment]
I'm not opposed to this type of change if it makes the UX cleaner.

My only feedback is that I would want it to be an append-unless-exists model. In other words, I wouldn't want the output to be `nist_rev: ['4', '4']` but rather `nist_rev: '4'`, no matter how many times you called `tag nist_rev: '4'`. I'm not sure duplicating values offers any value to the user and will likely require them to `#uniq` before using otherwise. Thoughts?

**Comment by aaronlippold** [comment]
Hi, I think @chris-rock  and @arlimus  actually resolved this in their ever kick butt response time 👍 

We can call this closed as of v.0.17.0 I think. https://github.com/chef/inspec/pull/1590/files

see: https://github.com/chef/inspec/blob/master/test/unit/dsl/objects_test.rb#L352-L363.

I then used it like this, https://github.com/aaronlippold/xccdf2inspec/blob/master/demo.rb#L31-L34

**Comment by adamleff** [comment]
@DeltaWhy do you agree that this functionality exists as you expect and we can close this issue?

**Comment by DeltaWhy** [comment]
That test doesn't look like it's testing the functionality in question. I would expect to see two tags added with the same name, and verify that there are two lines in the output of `to_ruby`.

It also looks like this doesn't match the DSL definition where `@tags` is a hash. I think this is the relevant method? https://github.com/chef/inspec/blob/master/lib/inspec/rule.rb#L88 Am I correct in thinking this file is part of the DSL and `lib/inspec/objects/control.rb` is in a sense the opposite, mapping objects back to DSL code? I'm not too familiar with this project yet and metaprogramming is hard to follow.

**Comment by miah** [comment]
__What’s happening? Why was this issue closed?__
This issue was closed due to some much needed review of legacy issues or issues that were spawned in older versions of InSpec, i.e. < v3.

__Why do I care?__
You would care about this if this was an issue that you are still seeing and/or feel needs to be addressed in the current version of InSpec.

__What do I need to do?__
If this issue is no longer important, no further action is necessary. However, if you think this is something that should be addressed, please open a new issue and refer to the original issue in the description.
